src/palindrom-network-channel.js

```diff
@@ -0,0 +1,350 @@
+                        upgradeURL,
+                        'WS'
+                    )
+                );
+            }
+        };
+    }
+    closeConnection() {
+        if (this._ws) {
+            this._ws.onclose = () => {};
+            this._ws.close();
+            this._ws = null;
+        }
+    }
+    /**
+     * @param {String} href
+     * @throws {Error} network error if occured
+     * @returns {Promise<Object>} fetched patch
+     * @see #_fetch
+     */
+    async getPatchUsingHTTP(href) {
+        // we don't need to try catch here because we want the error to be thrown at whoever calls getPatchUsingHTTP
+        const method = 'GET';
+        const data = await this._fetch(
+            method,
+            href,
+            'application/json-patch+json',
+            null,
+            true
+        );
+        this.onReceive(data, href, method);
+        return data;
+    }
+
+    _setRemoteUrl(remoteUrl) {
+        if (
+            this.remoteUrlSet &&
+            this.remoteUrl &&
+            this.remoteUrl != remoteUrl
+        ) {
+            const message = [
+                'Session lost.',
+                'Server replied with a different session ID than the already set one.',
+                'Possibly a server restart happened while you were working.',
+                'Please reload the page.',
+                'Previous session ID: ' + this.remoteUrl,
+                'New session ID: ' + remoteUrl
+            ].join('\n');
+
+            throw new PalindromError(message);
+        }
+        this.remoteUrlSet = true;
+        this.remoteUrl = new URL(remoteUrl, this.remoteUrl.href);
+    }
+
+    _handleLocationHeader(res) {
+        const location = res.headers.get('x-location') || res.headers.get('location');
+        if (location) {
+            this._setRemoteUrl(location);
+        }
+    }
+    /**
+     * Handles unsecessful HTTP requests
+     * @param error
+     */
+    async _handleFailureResponse(url, method, error) {
+        // no sufficient error information, we need to create on our own
+        var statusCode = -1;
+        var statusText = `An unknown network error has occurred. Raw message: ${
+            error.message
+        }`;
+        var reason = 'Maybe you lost connection with the server';
+        // log it for verbosity
+        console.error(error);
+
+        const message = [
+            statusText,
+            'statusCode: ' + statusCode,
+            'reason: ' + reason,
+            'url: ' + url,
+            'HTTP method: ' + method
+        ].join('\n');
+
+        this.onFatalError(
+            new PalindromConnectionError(message, CLIENT, url, method)
+        );
+    }
+
+    /**
+     * Internal method to perform HTTP Request.
+     * @param {String} method HTTP method to be used
+     * @param {String} [url=window.location] URL to send the request. If empty string, undefined or null given - the request will be sent to window location
```

Comment:
I think the part about the default value is incorrect. This method expects explicit `url`. Implicit use of `window.location` was removed few weeks ago on master: https://github.com/Palindrom/Palindrom/issues/219